src/cache.py

```
@@ -0,0 +1,103 @@
+
+async def get_group_participants(gid: str) -> List[str]:
+    doc  = groups.document(gid).get()
+    data = doc.to_dict() or {}
+    parts = data.get("participants")
+    if not parts:
+        from database import ChatDBClient
+        parts = ChatDBClient().get_participants(gid)
+        groups.document(gid).set({"participants": parts}, merge=True)
+    return parts
+
+class RedisCache:
```

Flake8 (7.2.0)

[error] expected 2 blank lines, found 1

(E302)